File: Euclid/Messages/Incoming/SEARCHBUSYFLATS.cs
using Euclid.Game;
using Euclid.Messages.Outgoing;
using Euclid.Network.Streams;
using Euclid.Storage.Database.Access;

namespace Euclid.Messages.Incoming
{
    class SEARCHBUSYFLATS : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            int page = 0;
            int maxResults = 11;

            if (request.GetArgumentAmount(",") > 0)
            {
                page = int.Parse(request.GetArgument(0, ",").Replace("/", ""));
                maxResults = int.Parse(request.GetArgument(1, ","));
            }

            player.Send(new BUSY_FLAT_RESULTS(RoomManager.Instance.ReplaceQueryRooms(RoomDao.GetPopularFlats(page / 11, maxResults))));
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/SEARCHBUSYFLATS.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.game.room.RoomManager;
import euclid.messages.IMessageEvent;
import euclid.messages.outgoing.BUSY_FLAT_RESULTS;
import euclid.network.streams.Request;
import euclid.storage.database.access.RoomDao;

public class SEARCHBUSYFLATS implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        int page = 0;
        int maxResults = 11;

        if (request.getArgumentAmount(",") > 0) {
            page = Integer.parseInt(request.getArgument(0, ",").replace("/", ""));
            maxResults = Integer.parseInt(request.getArgument(1, ","));
        }

        player.send(new BUSY_FLAT_RESULTS(RoomManager.getInstance().replaceQueryRooms(
                RoomDao.getPopularFlats(page / 11, maxResults))));
    }
}
